### PR TITLE
chore: add workflows to auto-close and lock stale issues and PRs

### LINKED ISSUE #683: Auto Close Stale Issues / PRs
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

To reduce the amount of things that are likely not relevant anymore, let's automate closing them. I like the [config](https://github.com/aws/aws-cdk/blob/963d1c7755e23ea819481724d7e8c78e31d82294/.github/workflows/close-stale-issues.yml) of what the AWS CDK repo has. It seems to be pretty straightforward.

### PR DESCRIPTION
**Pull request by xiehan**
When I was adding the `bug/has-workaround` label to tickets yesterday, I noticed there are a handful of issues labeled `waiting-on-answer` where we haven't heard back from the user in months. I figured we should automate posting a "warning" and then closing the issue after a set amount of time. I also added a workflow that locks threads after 30 days to encourage folks to create new issues rather than replying to old closed issues forever. Kudos to the Editor Experience team for giving me a template to work from!

Closes #683

### CODE REVIEW

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v6
+        with:
+          # For issues: post a "warning" message after 30 days, then close if another 30 days pass without a response. In another workflow, issues closed for 30 days will be locked.
+          stale-issue-message: "Hi there! 👋 We haven't heard from you in 30 days and would like to know if the problem has been resolved or if you still need help. If we don't hear from you before then, I'll auto-close this issue in 30 days."
+          close-issue-message: "I'm closing this issue because we haven't heard back in 60 days. ⌛️ If you still need help, feel free to comment or reopen the issue!"
+          days-before-issue-stale: 30
+          days-before-issue-close: 30
```
Comment (DanielMSchmidt) [inline .github/workflows/stale.yml:20]: This should be 60

**Review by DanielMSchmidt — CHANGES_REQUESTED** [review]

File: .github/workflows/stale.yml (hunk 1 above)
Comment (xiehan) [reply .github/workflows/stale.yml]: Good catch! The docs are a little bit confusing about this, but I think you're right.

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v6
+        with:
+          # For issues: post a "warning" message after 30 days, then close if another 30 days pass without a response. In another workflow, issues closed for 30 days will be locked.
+          stale-issue-message: "Hi there! 👋 We haven't heard from you in 30 days and would like to know if the problem has been resolved or if you still need help. If we don't hear from you before then, I'll auto-close this issue in 30 days."
+          close-issue-message: "I'm closing this issue because we haven't heard back in 60 days. ⌛️ If you still need help, feel free to comment or reopen the issue!"
+          days-before-issue-stale: 30
+          days-before-issue-close: 60
```
Comment (DanielMSchmidt) [inline .github/workflows/stale.yml:20]: Sorry, I misunderstood the API, it seems like adding the stale label is an activity so 30+30=60

```suggestion
          days-before-issue-close: 30
```

**Review by DanielMSchmidt — APPROVED** [review]
Sorry, I misunderstood the API :D

File: .github/workflows/lock.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: dessant/lock-threads@v3
+        with:
+          issue-comment: >
+            I'm going to lock this issue because it has been closed for 30 days. This helps our maintainers find and focus on the active issues.
+            If you've found a problem that seems similar to this, please open a new issue and complete the issue template so we can capture all the details necessary to investigate further.
+          issue-inactive-days: 30
+          pr-comment: >
+            I'm going to lock this pull request because it has been closed for 30 days. This helps our maintainers find and focus on the active issues.
```
Comment (mutahhir) [inline .github/workflows/lock.yml:22]: Nit: Still references issues here (and in the following line). Maybe change it to use 'pull requests' instead.

Comment (mutahhir) [reply .github/workflows/lock.yml:22]: That being said, I wonder if opening issues still makes more sense than open a new pull request. Feel free to ignore.

**Review by ansgarm — APPROVED** [review]

**Pull request merged**